packages/vx-axis/src/types.ts

```diff
@@ -1,30 +1,37 @@
+import { D3Scale, NumberLike, ScaleInput } from '@vx/scale';
 import { TextProps } from '@vx/text/lib/Text';
+import Orientation from './constants/orientation';
 
-export type AxisOrientation = 'top' | 'right' | 'bottom' | 'left';
+// In order to plot values on an axis, output of the scale must be number.
+// Some scales return undefined.
+export type AxisScaleOutput = number | NumberLike | undefined;
+
+/** A catch-all type for scales that are compatible with axis */
+export type AxisScale<Output extends AxisScaleOutput = AxisScaleOutput> =
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  D3Scale<Output, any, any>;
 
-export type FormattedValue = string | number | undefined;
+type FormattedValue = string | undefined;
 
-export type TickFormatter<ScaleInput> = (value: ScaleInput, tickIndex: number) => FormattedValue;
+export type TickFormatter<T> = (value: T, index: number) => FormattedValue;
 
-export type TickLabelProps<ScaleInput> = (val: ScaleInput, index: number) => Partial<TextProps>;
+export type TickLabelProps<T> = (value: T, index: number) => Partial<TextProps>;
 
 export type TickRendererProps = Partial<TextProps> & {
   x: number;
   y: number;
   formattedValue: FormattedValue;
 };
 
-export type SharedAxisProps<ScaleInput> = {
-  /** The class name applied to the outermost axis group element. */
-  axisClassName?: string;
+interface CommonProps<Scale extends AxisScale> {
```

Comment:
This is a new interface that is the shared props between `Axis` and `AxisRenderer`